### PR TITLE
Bench 328

### PR DESCRIPTION
**Pull request by martintaylor1635**

### CODE REVIEW

File: src/main/java/com/answerdigital/answerking/service/TagService.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        this.tagRepository = tagRepository;
+        this.productService = productService;
+    }
+
+    public TagResponse addTag(final TagRequest tagRequest) {
+        if (tagRepository.existsByName(tagRequest.name())) {
+            throw getCustomException(TAGS_ALREADY_EXIST, tagRequest.name());
+        }
+
+        final List<Product> products = productService.findAllProductsInListOfIds(tagRequest.productIds());
```
Comment (AndyFlintAnswerDigital) [inline src/main/java/com/answerdigital/answerking/service/TagService.java:38]: This needs to be able to return a 404 if one or more productIds do not exist. We're imagining an approach where we'd compare the IDs in the request with the IDs of the returned objects, and we can then get the difference of the two lists and return that list in an error message to the front end.

File: src/test/java/com/answerdigital/answerking/service/TagServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            .when(productService)
+            .findAllProductsInListOfIds(anyList());
+        doReturn(expectedResponse)
+            .when(tagRepository)
+            .save(any(Tag.class));
+
+        final TagResponse tagResponse = tagService.addTag(tagRequest);
+
+        // then
+        assertTagVsResponseEquality(expectedResponse, tagResponse);
```
Comment (AndyFlintAnswerDigital) [inline src/test/java/com/answerdigital/answerking/service/TagServiceTest.java:101]: Needs verifys

**Review by AndyFlintAnswerDigital — CHANGES_REQUESTED** [review]

File: src/main/java/com/answerdigital/answerking/service/TagService.java (hunk 1 above)
Comment (AndyFlintAnswerDigital) [reply src/main/java/com/answerdigital/answerking/service/TagService.java:38]: UPDATE - We'll make this a separate ticket as it'll require API spec changes.

File: src/test/java/com/answerdigital/answerking/service/TagServiceTest.java (hunk 2 above)
Comment (martintaylor1635) [reply src/test/java/com/answerdigital/answerking/service/TagServiceTest.java:103]: ✔️ Fixed

File: src/main/java/com/answerdigital/answerking/service/TagService.java (hunk 1 above)
Comment (martintaylor1635) [reply src/main/java/com/answerdigital/answerking/service/TagService.java:38]: ❌ Won't fix on the basis of separating this into a separate ticket.

**Pull request merged**